### ISSUE TITLE
Spline lineage harvester fails on Spark literals containing array of structs

### ISSUE THREAD

**Issue by yruslan** [AbsaOSS/spline#296]
Spark version: 2.4.3
Spline version: 0.3.8

Stack trace:

```
19/07/08 15:45:22 INFO DAGScheduler: Job 56 finished: parquet at DynamicConformanceJob.scala:194, took 20.951173 s
19/07/08 15:45:22 INFO FileFormatWriter: Write Job 7a59d4fb-899d-4d18-9683-46622520bd88 committed.
19/07/08 15:45:22 INFO FileFormatWriter: Finished processing stats for write job 7a59d4fb-899d-4d18-9683-46622520bd88.
19/07/08 15:45:23 WARN ExecutionListenerManager: Error executing query execution listener
java.lang.ClassCastException: org.apache.spark.sql.catalyst.expressions.GenericInternalRow cannot be cast to org.apache.spark.sql.catalyst.util.ArrayData
	at org.apache.spark.sql.catalyst.util.GenericArrayData.getArray(GenericArrayData.scala:79)
	at org.apache.spark.sql.catalyst.InternalRow$$anonfun$getAccessor$13.apply(InternalRow.scala:141)
	at org.apache.spark.sql.catalyst.InternalRow$$anonfun$getAccessor$13.apply(InternalRow.scala:141)
	at org.apache.spark.sql.catalyst.util.ArrayData.toArray(ArrayData.scala:178)
	at za.co.absa.spline.core.harvester.ExpressionConverter$.za$co$absa$spline$core$harvester$ExpressionConverter$$getLiteralValue(componentCreators.scala:143)
	at za.co.absa.spline.core.harvester.ExpressionConverter.convert(componentCreators.scala:166)
	at za.co.absa.spline.core.harvester.ExpressionConverter$$anonfun$convert$4.apply(componentCreators.scala:178)
	at za.co.absa.spline.core.harvester.ExpressionConverter$$anonfun$convert$4.apply(componentCreators.scala:178)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at za.co.absa.spline.core.harvester.ExpressionConverter.convert(componentCreators.scala:178)
	at za.co.absa.spline.core.harvester.ExpressionConverter$$anonfun$convert$4.apply(componentCreators.scala:178)
	at za.co.absa.spline.core.harvester.ExpressionConverter$$anonfun$convert$4.apply(componentCreators.scala:178)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at za.co.absa.spline.core.harvester.ExpressionConverter.convert(componentCreators.scala:178)
	at za.co.absa.spline.core.harvester.ExpressionConverter$$anonfun$convert$5.apply(componentCreators.scala:191)
	at za.co.absa.spline.core.harvester.ExpressionConverter$$anonfun$convert$5.apply(componentCreators.scala:191)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.immutable.List.map(List.scala:296)
	at za.co.absa.spline.core.harvester.ExpressionConverter.convert(componentCreators.scala:191)
	at za.co.absa.spline.core.harvester.ExpressionConverter.convert(componentCreators.scala:160)
	at za.co.absa.spline.core.harvester.ProjectionNodeBuilder$$anonfun$2.apply(operationBuilders.scala:186)
	at za.co.absa.spline.core.harvester.ProjectionNodeBuilder$$anonfun$2.apply(operationBuilders.scala:186)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at za.co.absa.spline.core.harvester.ProjectionNodeBuilder.build(operationBuilders.scala:186)
	at za.co.absa.spline.core.harvester.ProjectionNodeBuilder.build(operationBuilders.scala:179)
	at za.co.absa.spline.core.harvester.DataLineageBuilder$$anonfun$2.apply(DataLineageBuilder.scala:50)
	at za.co.absa.spline.core.harvester.DataLineageBuilder$$anonfun$2.apply(DataLineageBuilder.scala:50)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.immutable.List.map(List.scala:296)
	at za.co.absa.spline.core.harvester.DataLineageBuilder.buildLineage(DataLineageBuilder.scala:50)
	at za.co.absa.spline.core.harvester.QueryExecutionEventHandler.onSuccess(QueryExecutionEventHandler.scala:43)
	at za.co.absa.spline.core.listener.SplineQueryExecutionListener$$anonfun$onSuccess$1.apply(SplineQueryExecutionListener.scala:37)
	at za.co.absa.spline.core.listener.SplineQueryExecutionListener$$anonfun$onSuccess$1.apply(SplineQueryExecutionListener.scala:37)
	at scala.Option.foreach(Option.scala:257)
	at za.co.absa.spline.core.listener.SplineQueryExecutionListener.onSuccess(SplineQueryExecutionListener.scala:37)
	at org.apache.spark.sql.util.ExecutionListenerManager$$anonfun$onSuccess$1$$anonfun$apply$mcV$sp$1.apply(QueryExecutionListener.scala:124)
	at org.apache.spark.sql.util.ExecutionListenerManager$$anonfun$onSuccess$1$$anonfun$apply$mcV$sp$1.apply(QueryExecutionListener.scala:123)
	at org.apache.spark.sql.util.ExecutionListenerManager$$anonfun$org$apache$spark$sql$util$ExecutionListenerManager$$withErrorHandling$1.apply(QueryExecutionListener.scala:145)
	at org.apache.spark.sql.util.ExecutionListenerManager$$anonfun$org$apache$spark$sql$util$ExecutionListenerManager$$withErrorHandling$1.apply(QueryExecutionListener.scala:143)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.generic.TraversableForwarder$class.foreach(TraversableForwarder.scala:35)
	at scala.collection.mutable.ListBuffer.foreach(ListBuffer.scala:45)
	at org.apache.spark.sql.util.ExecutionListenerManager.org$apache$spark$sql$util$ExecutionListenerManager$$withErrorHandling(QueryExecutionListener.scala:143)
	at org.apache.spark.sql.util.ExecutionListenerManager$$anonfun$onSuccess$1.apply$mcV$sp(QueryExecutionListener.scala:123)
	at org.apache.spark.sql.util.ExecutionListenerManager$$anonfun$onSuccess$1.apply(QueryExecutionListener.scala:123)
	at org.apache.spark.sql.util.ExecutionListenerManager$$anonfun$onSuccess$1.apply(QueryExecutionListener.scala:123)
	at org.apache.spark.sql.util.ExecutionListenerManager.readLock(QueryExecutionListener.scala:156)
	at org.apache.spark.sql.util.ExecutionListenerManager.onSuccess(QueryExecutionListener.scala:122)
	at org.apache.spark.sql.DataFrameWriter.runCommand(DataFrameWriter.scala:678)
	at org.apache.spark.sql.DataFrameWriter.saveToV1Source(DataFrameWriter.scala:285)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:271)
	at org.apache.spark.sql.DataFrameWriter.save(DataFrameWriter.scala:229)
	at org.apache.spark.sql.DataFrameWriter.parquet(DataFrameWriter.scala:566)
	at za.co.absa.enceladus.conformance.DynamicConformanceJob$.processResult(DynamicConformanceJob.scala:194)
	at za.co.absa.enceladus.conformance.DynamicConformanceJob$.main(DynamicConformanceJob.scala:123)
	at za.co.absa.enceladus.conformance.DynamicConformanceJob.main(DynamicConformanceJob.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:849)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:167)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:195)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:924)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:933)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
```

**Comment by wajda** [comment]
@yruslan , @GeorgiChochov , @benedeki 
Fixed in Spline 0.3.9. Can you guys test it please?

**Comment by yruslan** [comment]
Confirm it works on my local machine.
